File: USACO/Silver/sort.java
import java.util.*;
import java.io.*;

public class sort
{
	static class Event implements Comparable<Event> {
		int ind, val;
		public Event(int i, int v) {
			ind = i;
			val = v;
		}
		public int compareTo(Event a) {
			return this.val == a.val ? this.ind - a.ind : this.val - a.val;
		}
	}
	
	public static void main(String[] args) throws IOException {	
	    BufferedReader br = new BufferedReader(new FileReader("sort.in"));
		PrintWriter pw = new PrintWriter(new FileWriter("sort.out"));
		
		int n = Integer.parseInt(br.readLine());
		ArrayList<Event> arr = new ArrayList<Event>();
		for (int i = 0; i < n; i++) {
			arr.add(new Event(i, Integer.parseInt(br.readLine())));
		}
		Collections.sort(arr);
		
		int ans = 0;
		for (int i = 0; i < n; i++) {
			ans = Math.max(ans, arr.get(i).ind - i);
		}
		pw.println(ans + 1);
		pw.close();
	}
}

File: USACO/Silver/sort.cpp
#include <bits/stdc++.h>
#define MAXN 100005
#define MOD 1000000007
#define ll long long
#define mp make_pair
#define sz(x) (int) (x).size()
#define pb push_back

using namespace std;

struct Event { int ind, val; };

bool operator<(Event a, Event b) { return a.val == b.val ? a.ind < b.ind : a.val < b.val; }

Event arr[MAXN];

int main() {
	freopen("sort.in", "r", stdin);
	freopen("sort.out", "w", stdout);
	int n; cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> arr[i].val;
		arr[i].ind = i;
	}
	sort(arr, arr + n);
	
	int ans = 0;
	for (int i = 0; i < n; i++) {
		ans = max(ans, arr[i].ind - i);
	}
	cout << ans + 1 << '\n';
	return 0;
}